src/ecs/resources/input.rs

```diff
@@ -22,14 +22,52 @@ impl<'a> Iterator for PressedKeysIterator<'a> {
     }
 }
 
+/// Iterator that iterates through all currently pressed down mouse buttons
+pub struct PressedMouseButtonsIterator<'a> {
```

Comment:
Hmm, this name seems quite long and cumbersome. Perhaps we could trim this down to `PressedMouseButtons`, following the pattern used in the standard library (e.g. `std::collections::hash_map::Keys` as opposed to `std::collections::hash_map::KeysIterator`)?